libs/gltfio/CMakeLists.txt

```diff
@@ -104,29 +103,39 @@ add_custom_command(OUTPUT ${DUMMY_SRC} COMMAND echo "//" > ${DUMMY_SRC})
 
 # ==================================================================================================
-# Include and target definitions
+# Build the "core" library (no filamat)
 # ==================================================================================================
 include_directories(${PUBLIC_HDR_DIR} ${RESOURCE_DIR})
+link_libraries(math utils filament cgltf stb geometry gltfio_resources)
 
-add_library(${TARGET} STATIC ${PUBLIC_HDRS} ${SRCS})
+add_library(gltfio_core STATIC ${PUBLIC_HDRS} ${SRCS})
 
-target_link_libraries(${TARGET} PUBLIC math utils filamat filament cgltf stb geometry gltfio_resources)
+target_include_directories(gltfio_core PUBLIC ${PUBLIC_HDR_DIR})
 
-target_include_directories(${TARGET} PUBLIC ${PUBLIC_HDR_DIR})
+if (NOT WEBGL AND NOT ANDROID)
 
-# ==================================================================================================
-# Compiler flags
-# ==================================================================================================
-target_compile_options(${TARGET} PRIVATE -Wno-deprecated-register)
+    # ==================================================================================================
+    # Link the core library against filamat to create the "full" library
+    # ==================================================================================================
+    add_library(${TARGET} STATIC ${PUBLIC_HDRS} src/MaterialGenerator.cpp)
+    target_link_libraries(${TARGET} PUBLIC filamat gltfio_core)
+    target_include_directories(${TARGET} PUBLIC ${PUBLIC_HDR_DIR})
 
-if (MSVC OR CLANG_CL)
-    target_compile_options(${TARGET} PRIVATE $<$<CONFIG:Release>:/fp:fast>)
-else()
-    target_compile_options(${TARGET} PRIVATE $<$<CONFIG:Release>:-ffast-math>)
-endif()
+    # ==================================================================================================
+    # Compiler flags
+    # ==================================================================================================
+    target_compile_options(${TARGET} PRIVATE -Wno-deprecated-register)
 
-# ==================================================================================================
-# Installation
-# ==================================================================================================
-install(TARGETS ${TARGET} ARCHIVE DESTINATION lib/${DIST_DIR})
-install(DIRECTORY ${PUBLIC_HDR_DIR}/gltfio DESTINATION include)
+    if (MSVC OR CLANG_CL)
+        target_compile_options(${TARGET} PRIVATE $<$<CONFIG:Release>:/fp:fast>)
+    else()
+        target_compile_options(${TARGET} PRIVATE $<$<CONFIG:Release>:-ffast-math>)
+    endif()
+
+    # ==================================================================================================
+    # Installation
+    # ==================================================================================================
+    install(TARGETS ${TARGET} ARCHIVE DESTINATION lib/${DIST_DIR})
```

Comment:
If I'm not mistaken, `gltfio_core` will also need to be installed to `lib/gltfio_core.a` so clients can link against it as well.

Reply:
Ah thanks, fixed this.  Now, if only I can get the presubmit to pass...